external-crates/move/crates/move-vm-runtime/src/execution/dispatch_tables.rs

```diff
@@ -1109,43 +884,41 @@ impl DepthFormula {
     }
 }
 
-impl CachedDatatype {
-    pub fn get_struct(&self) -> PartialVMResult<&StructType> {
-        match &self.datatype_info {
-            Datatype::Struct(struct_type) => Ok(struct_type),
-            x @ Datatype::Enum(_) => Err(PartialVMError::new(
-                StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
-            )
-            .with_message(format!("Expected struct type but got {:?}", x))),
+// ------------------------------------------------------------------------
+// Other Impls
+// ------------------------------------------------------------------------
+
+impl PackageVirtualTable {
+    pub fn new() -> Self {
+        Self {
+            functions: BTreeMap::new(),
+            types: BTreeMap::new(),
         }
     }
+}
 
-    pub fn get_enum(&self) -> PartialVMResult<&EnumType> {
-        match &self.datatype_info {
-            Datatype::Enum(enum_type) => Ok(enum_type),
-            x @ Datatype::Struct(_) => Err(PartialVMError::new(
-                StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR,
-            )
-            .with_message(format!("Expected enum type but got {:?}", x))),
-        }
+impl VirtualTableKey {
+    pub fn to_string(&self) -> PartialVMResult<String> {
+        let inner_name = self.inner_pkg_key.to_string()?;
+        Ok(format!("{}::{}", self.package_key, inner_name))
     }
+}
 
-    pub fn datatype_key(&self) -> VirtualTableKey {
-        let module_name = self.module_key;
-        let member_name = self.member_key;
-        VirtualTableKey {
-            package_key: *self.runtime_id.address(),
-            inner_pkg_key: IntraPackageKey {
-                module_name,
-                member_name,
-            },
-        }
+impl IntraPackageKey {
+    pub fn to_string(&self) -> PartialVMResult<String> {
+        let module_name = string_interner().resolve_string(&self.module_name, "module name")?;
+        let member_name = string_interner().resolve_string(&self.member_name, "member name")?;
+        Ok(format!("{}::{}", module_name, member_name))
     }
 }
 
-impl CachedDatatype {
-    pub fn type_param_constraints(&self) -> impl ExactSizeIterator<Item = &AbilitySet> {
-        self.type_parameters.iter().map(|param| &param.constraints)
+// -------------------------------------------------------------------------------------------------
+// Default
+// -------------------------------------------------------------------------------------------------
+
+impl Default for PackageVirtualTable {
```

Comment:
Have I ever mentioned how much I hate `Default` and in particular Rust's clippy lint about this?...